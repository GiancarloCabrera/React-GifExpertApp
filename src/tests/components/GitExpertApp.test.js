import React from "react"
import { shallow } from "enzyme"
import GitExpertApp from "../../components/GifExpertApp"

describe('Pruebas en el componente GitExpertApp', () => { 
    test('Debe mostrarse correctamente', () => { 
        const app = shallow(<GitExpertApp/>)    
        expect( app ).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de categorias', () => { 
        //Creo un array el cual se le pasara a mi componente
        const categories = ['One Punch', 'Dragon Ball'];
        const app = shallow(<GitExpertApp
            defaultCategories={ categories }
        />);  

        //Hago el snapshot
        expect( app ).toMatchSnapshot();

        //Evaluo si el componente GifGrid esta siendo llamado el mismo numero de veces que el categories
        expect( app.find('GifGrid').length).toBe( categories.length );
    });
 })