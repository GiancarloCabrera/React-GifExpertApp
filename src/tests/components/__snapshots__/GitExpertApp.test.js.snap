// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GitExpertApp Debe de mostrar una lista de categorias 1`] = `
<div>
  <h2>
    GitExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</div>
`;

exports[`Pruebas en el componente GitExpertApp Debe mostrarse correctamente 1`] = `
<div>
  <h2>
    GitExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
